// next/dist/server/web/adapter.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
exports.adapter = adapter;
var _error = require("./error");
var _utils = require("./utils");
var _fetchEvent = require("./spec-extension/fetch-event");
var _request = require("./spec-extension/request");
var _response = require("./spec-extension/response");
var _fetchEvent1 = require("./spec-compliant/fetch-event");
async function adapter(params) {
    const url = params.request.url.startsWith('/') ? `https://${params.request.headers.host}${params.request.url}` : params.request.url;
    const request = new NextRequestHint(url, {
        geo: params.request.geo,
        headers: (0, _utils).fromNodeHeaders(params.request.headers),
        ip: params.request.ip,
        method: params.request.method,
        nextConfig: params.request.nextConfig,
        page: params.request.page
    });
    const event = new _fetchEvent.NextFetchEvent(request);
    const original = await params.handler(request, event);
    return {
        response: original || _response.NextResponse.next(),
        waitUntil: Promise.all(event[_fetchEvent1.waitUntilSymbol])
    };
}
class NextRequestHint extends _request.NextRequest {
    constructor(input, init = {
    }){
        super(input, init);
    }
    get request() {
        throw new _error.DeprecationError();
    }
    respondWith() {
        throw new _error.DeprecationError();
    }
    waitUntil() {
        throw new _error.DeprecationError();
    }
}
